### DISCUSSION TITLE
What would it take to make a version of this library for sequel?

### DISCUSSION THREAD

**Discussion by pboling** [ClosureTree/closure_tree#423] [Ideas]
Or is there one already that I missed?

**Comment by seuros** [comment]
I don't know  i didnt see this.

Did you find a solution ?

**Reply by pboling** [reply]
No, I have only just begun using Sequel (not even launched into prod yet), and I had an idea of a direction I might want to take that usage which would have involved a closure_tree such as this project provides, but I've already forgotten what that was.

**Comment by timuckun** [comment]
I looked at the source code a bit (just a couple of files)  and it uses arel to construct queries. It also used Activesupport::OrderedHash in the file I looked at.  

 Having said that it does require activerecord in the gemspec so it might actually use AR in other places in the code. 
 
 If it's using only Arel to construct queries you might be able to modify it slightly to make it work with sequel. I don't think you could get rid of the arel dependency though.

**Reply by seuros** [reply]
Closure Tree is a pattern.  You can have it in AR, Sequel or even raw SQL.  The trick is to have 2 tables .

If someone build the Sequel version, we could add it the gem to the Org.  

I tried to do it in Rust https://github.com/ClosureTree/closure-tree-rs , but i gave up cuz rust ORM was not civilized.
